Source/core/Proxy.h

```diff
@@ -1094,17 +1087,15 @@ namespace Core {
             // -----------------------------------------------------
             HAS_MEMBER(Relinquish, hasRelinquish);
 
-            typedef hasRelinquish<ELEMENT, void (ELEMENT::*)()> TraitRelinquish;
-
-            template <typename TYPE>
-            inline typename Core::TypeTraits::enable_if<ProxyObjectType<TYPE>::TraitRelinquish::value, void>::type
+            template <typename TYPE=ELEMENT>
+            inline typename Core::TypeTraits::enable_if<hasRelinquish<TYPE, void (TYPE::*)()>::value, void>::type
             __Relinquish()
             {
                 ELEMENT::Relinquish();
             }
 
             template <typename TYPE>
```

Comment:
You forgot the default argument, You wiill notice any when you correct the usage and remove the template param from the call :)